swap/src/alice.rs

```diff
@@ -105,15 +106,24 @@ pub async fn swap(
     Ok(())
 }
 
-fn new_swarm(listen: Multiaddr) -> Result<Swarm> {
+fn new_swarm(listen: Multiaddr, port: u16) -> Result<Swarm> {
     use anyhow::Context as _;
 
     let behaviour = Alice::default();
 
     let local_key_pair = behaviour.identity();
     let local_peer_id = behaviour.peer_id();
 
-    let transport = transport::build(local_key_pair)?;
+    let transport = {
+        #[cfg(feature = "tor")]
+        {
+            transport::build(local_key_pair, Some((listen.clone(), port)))?
+        }
+        #[cfg(not(feature = "tor"))]
+        {
+            transport::build(local_key_pair)?
+        }
+    };
```

Comment:
```suggestion
    #[cfg(feature = "tor")]
    let transport = transport::build(local_key_pair, Some((listen.clone(), port)))?
    #[cfg(not(feature = "tor"))]
    let transport = transport::build(local_key_pair)?
```

Reply:
Maybe I did something wrong but that did not work for me...